### PR TITLE
fixes clowns and mimes losing their genes

### PR DESCRIPTION
**Pull request by NewyearnewmeUwu**
## About the PR 
Title says it all. Traits now apply genes upon being added, so clowns and mimes don't lose their funny genes after being cloned!

## Why's this needed? 
This fixes a major bug that I myself reported in #11935 almost a year ago!

## Changelog 

```changelog
(u)NewyearnewmeUwu
(+)Fixed clowns and mimes losing their career genes after cloning.
```

### CODE REVIEW

**Comment by MeggalBozale** [comment]
Looks good, sounds good too. No more mimes popping out the cloner talking

**Review by MeggalBozale — APPROVED** [review]
Once-over LGTM, in-game test functions properly. Maybe a thing or two that could be improved code wise that I wouldn't know about

File: code/datums/traits.dm [hunk 1]
```diff
@@ -616,6 +616,25 @@ ABSTRACT_TYPE(/datum/trait/job)
...
+		owner.bioHolder?.AddEffect("clumsy", magical = TRUE)
+
+/datum/trait/job/mime
+	name = "Mime Training"
+	desc = "..."
+	id = "training_mime"
+
+	onAdd(var/mob/owner)
+		owner.bioHolder.AddEffect("mute", magical = TRUE)
+		owner.bioHolder.AddEffect("blankman", magical = TRUE)
```
Comment (kbsmilk) [inline code/datums/traits.dm:636]:
```suggestion
		owner.bioHolder?.AddEffect("mute", magical = TRUE)
		owner.bioHolder?.AddEffect("blankman", magical = TRUE)
```
Looks like the other traits check for this too

**Review by kbsmilk — APPROVED** [review]

**Review by frawhst — APPROVED** [review]
lgtm

**Pull request merged**